### PR TITLE
Aleksei/extension queue length check

### PR DESCRIPTION
**Pull request by iambeone**
Closes #ISSUE

**Description:**

1. Checking extension queue size before open ActionModal
2. Canceling unsigned transactions in extension on the cancel button pressed

Thank you! 🚀

---

For contributor:

- [ ] Added changes entries. Run `yarn changelog` for a guided process.
- [ ] Reviewed `Files changed` in the github PR explorer
- [ ] Attach screenshots of the UI components on the PR description (if applicable)
- [ ] Scope of work approved for big PRs

For reviewer:

- [ ] Manually tested the changes on the UI

### CODE REVIEW

File: src/ActionModal/components/ActionModal.vue [hunk 1]
```diff
@@ -537,20 +538,35 @@ export default {
...
     confirmModalOpen() {
       let confirmResult = false
-      if (this.session.currrentModalOpen) {
+      if (this.session.currrentModalOpen || this.queueNotEmpty) {
         confirmResult = window.confirm(
           "You are in the middle of creating a transaction. Are you sure you want to cancel this action and start a new one?"
         )
         if (confirmResult) {
-          this.session.currrentModalOpen.close()
+          if (!this.queueNotEmpty) {
```
Comment (faboweb) [inline src/ActionModal/components/ActionModal.vue]: avoid double negation. hard to read. use `queueEmpty` instead.

**Comment by faboweb** [comment]
Needs extension release first

**Comment by faboweb** [comment]
@iambeone changelog missing

**Comment by iambeone** [comment]
@faboweb it's there. The first changed file in Files changed tab

**Comment by faboweb** [comment]
Missing coverage still

**Comment by iambeone** [comment]
It was adjusted ) Previously this block was removed and now it's back. Line that are missing couldn't be tested or could be just "tested".

**Comment by faboweb** [comment]
cover https://codecov.io/gh/luniehq/lunie/compare/15d001d181a32711229e65f13672ab39d6b199ca...69e1607c977bf2aa150ece14c5ddba7cb8128b14/diff#D1-566
cover https://codecov.io/gh/luniehq/lunie/compare/15d001d181a32711229e65f13672ab39d6b199ca...69e1607c977bf2aa150ece14c5ddba7cb8128b14/diff#D7-225

**Comment by Bitcoinera** [comment]
I have done some work on this (apollo query coverage). Currently still broken. 

Should I push anyway?

**Comment by iambeone** [comment]
I was thinking, i'm going crazy, cause I have fixed this coverage for sure... But the fix is here - https://github.com/luniehq/lunie/pull/3637/files... Updated this pr

**Comment by Bitcoinera** [comment]
> Updated this pr

Amazing. Let's merge this then!

**Comment by iambeone** [comment]
Don't merge it. Need to do some fixed for identifying missing extension answer

File: tests/unit/specs/components/wallet/PageTransactions.spec.js [hunk 2]
```diff
@@ -187,6 +187,65 @@ describe(`PageTransactions`, () => {
...
+              memo: "",
+              fee: { amount: "4141", denom: "ATOM" },
+              group: "staking"
+            }
+          }
+        }
+      }
+    )
+
+    expect(result.transactions.length).toBeGreaterThan(0)
```
Comment (faboweb) [inline tests/unit/specs/components/wallet/PageTransactions.spec.js]: more important to test the actual content of result.transactions to see if values get created correctly.

File: tests/unit/specs/components/wallet/PageTransactions.spec.js [hunk 3]
```diff
@@ -187,6 +187,65 @@ describe(`PageTransactions`, () => {
...
+          height: 1248479,
+          timestamp: "2019-07-31T09:22:23.054Z",
+          memo: "",
+          fee: { amount: "4141", denom: "ATOM" },
+          group: "staking"
+        }
+      ]
+    })
+
+    expect(result.length).toBeGreaterThan(0)
```
Comment (faboweb) [inline tests/unit/specs/components/wallet/PageTransactions.spec.js]: more important to test the actual content of result.transactions to see if values get created correctly.

File: src/ActionModal/components/ActionModal.vue [hunk 4]
```diff
@@ -562,6 +562,7 @@ export default {
       const queue = await this.actionManager.getSignQueue(
         this.selectedSignMethod
       )
+      console.log(queue)
```
Comment (Bitcoinera) [inline src/ActionModal/components/ActionModal.vue]: Please, don't forget to remove this line

File: tests/unit/specs/components/wallet/PageTransactions.spec.js (hunk 3 above)
Comment (iambeone) [reply tests/unit/specs/components/wallet/PageTransactions.spec.js]: Fixed

**Comment by iambeone** [comment]
Now, it's ok, could be merged

File: tests/unit/specs/components/wallet/PageTransactions.spec.js [hunk 5]
```diff
@@ -224,27 +224,28 @@ describe(`PageTransactions`, () => {
...
-          fee: { amount: "4141", denom: "ATOM" },
-          group: "staking"
-        }
-      ]
+      transactions: [newTransaction]
     })
-
-    expect(result.length).toBeGreaterThan(0)
+    // adjusting result
+    newTransaction.value = JSON.parse(newTransaction.value)
```
Comment (faboweb) [inline tests/unit/specs/components/wallet/PageTransactions.spec.js]: XD

**Comment by Bitcoinera** [comment]
Let's merge this! 💪

**Pull request merged**